module/assets/helper.js

```diff
@@ -0,0 +1,416 @@
+          reqData.toUserProfile.userId && checkUsersRolesIsIdentical
+        ) {
+          //get Fromuser details from user Extension
+
+          let fromUserData = await userExtensionsHelper.userExtensionDocument(
+            fromFindQuery
+          );
+          //get Touser details from user Extension
+          let toFindQuery = {
+            userId: reqData.toUserProfile.userId,
+          };
+          let toUserData = await userExtensionsHelper.userExtensionDocument(
+            toFindQuery
+          );
+          let allAssetsData = fromUserData?.platformRoles;
+          let typeOfAssetsToMove = reqData.assetInformation.objectType;
+          let checkAssetInformation =
+            reqData.hasOwnProperty("assetInformation");
+
+          //condition for if there is no object type
+          if (!checkAssetInformation) {
+            if (
+              reqData.toUserProfile.roles.includes(
+                constants.common.CONTENT_CREATOR
+              )
+            ) {
+              //filter for solution updates
+              let solutionFilter = {
+                author: reqData.fromUserProfile.userId,
+              };
+              let updateSolutions = {
+                $set: {
+                  author: reqData.toUserProfile.userId,
+                  creator: reqData.toUserProfile.firstname,
+                },
+              };
+              let solutionLicenseFilter = {
+                author: reqData.fromUserProfile.userId,
+                license: { $exists: true, $ne: "" },
+              };
+              let updateSolutionsLicense = {
+                $set: {
+                  "license.author": reqData.toUserProfile.firstName,
+                  "license.creator": reqData.toUserProfile.firstName,
+                },
+              };
+              let updateUserSolutions = [
+                solutionsHelper.updateMany(solutionFilter, updateSolutions),
+                solutionsHelper.updateMany(
+                  solutionLicenseFilter,
+                  updateSolutionsLicense
+                ),
+              ];
+              updateUserSolutionsDataResult = await Promise.all(
+                updateUserSolutions
+              );
+            } else {
+              if (
+                reqData.toUserProfile.roles.includes(
+                  constants.common.PROGRAM_MANAGER
+                ) ||
+                reqData.toUserProfile.roles.includes(
+                  constants.common.PROGRAM_DESIGNER
+                )
+              ) {
+                if (toUserData) {
+                  const updateQueries =
+                    await assetService.generateUpdateOperations(
+                      fromUserData,
+                      reqData,
+                      toUserData,
+                      allAssetsData
+                    );
+                  let updateProgram =
+                    await userExtensionsHelper.findOneandUpdate(updateQueries);
```

Comment:
rename the function if you are doinf bulk operation then its bulk operation only

Reply:
resolved